lib/ramble/ramble/workspace/workspace.py

```diff
@@ -1148,6 +1148,62 @@ def create_mirror(self, mirror_root):
         self.input_mirror_cache = ramble.caches.MirrorCache(self.input_mirror_path)
         self.software_mirror_cache = ramble.caches.MirrorCache(self.software_mirror_path)
 
+    def simplify(self):
+        # First drop unused experiment templates from app dict so environments aren't rendered
+        app_dict = ramble.config.config.get_config('applications',
+                                                   scope=self.ws_file_config_scope_name())
+
+        # Build experiment sets to determine which templates never get used
+        self.software_environments = \
+            ramble.software_environments.SoftwareEnvironments(self)
+        experiment_set = self.build_experiment_set()
+
+        for _, app_inst in experiment_set.template_experiments():
+            if app_inst.is_template and not app_inst.generated_experiments:
+                app = app_inst.expander.application_name
+                wl = app_inst.expander.workload_name
+                exp = app_inst.expander.experiment_name
+
+                try:
+                    app_dict[app]['workloads'][wl]['experiments'].pop(exp)
```

Comment:
`'workloads'` --> `namespace.workload`

`'experiments'` --> `namespace.experiment`

(And apply to below as well)